Feat/labels: Adding labels and annotations to pods

jachinte opened the pull request:
This pull request implements the feature I proposed in #32 

As you suggested, I've added some assertions to test the methods and attributes I added, and verified that they're 100% covered using the Jacoco report. I didn't add tests for verifying that deployed pods are actually labelled and annotated because I assumed this is Kubernete's job, but please let me know if you think this is necessary or whether I've forgotten to verify something.

I added some details on labels and annotations in the getting started guide, but I'm unsure whether this covers the CRD / CR breakdown you mentioned in #32. Please feel free to suggest edits and will gladly include them in this PR.

jachinte commented:
@AbdelrhmanHamouda Could you please run the CI pipeline again? It seems like there was an internal error -- unrelated to the code included in this PR.

AbdelrhmanHamouda commented:
Hello @jachinte, 
I will fully review the PR and will rerun CI pipelines tomorrow. I know this is taking a minute but I appreciate your patience.

AbdelrhmanHamouda reviewed (changes requested):
Hi @jachinte, 
Only few comments to address and minor things to tweak and then we are good to merge. 

I also kindly ask you to completely squash your commits into 1 or 2 max "feat(#32): ...." commits. The  CHANGELOG.md for the project is automated and all those feat:... commits will show as unique features to the project which is not the intended behaviour.

AbdelrhmanHamouda commented:
Regarding CI, I have reran it and it is fine. The failing flow is related to `pull_request` event which makes it so forks doesn't have access to repo secrets.

jachinte commented on gradle/dependencies.gradle:
```diff
@@ -48,5 +48,13 @@ dependencies {
     testImplementation("com.github.stefanbirkner:system-lambda:$systemLambdaVersion")
     testImplementation("io.fabric8:kubernetes-server-mock:$k8sClient")
 
+    if (System.getProperty("os.name").equals("Mac OS X")) {
```
You were right, it was an issue with k3s/k3d. I created the cluster using Kind and tests ran smoothly. I've reverted this change.

jachinte commented:
Hi @AbdelrhmanHamouda,

Thank you for reviewing the code and for suggesting the changes; they were indeed on point! I've updated the code as suggested and squashed all the commits into a single `feat` commit.

AbdelrhmanHamouda commented:
LGTM. 
Resolve #32
